### ISSUE TITLE
[Julia 1.7] Segfault when testing Knitro's API

### ISSUE THREAD

**Issue by frapac** [jump-dev/KNITRO.jl#208]
The [fourth testset](https://github.com/jump-dev/KNITRO.jl/blob/master/test/knitroapi.jl#L450-L541) returns a non-deterministic segfault in Julia 1.7, related to the GC. 

```
signal (11): Segmentation fault
in expression starting at /home/frapac/.julia/dev/KNITRO/tmp/segfault.jl:92
jl_gc_pool_alloc at /buildworker/worker/package_linux64/build/src/gc.c:1217
jl_gc_alloc_ at /buildworker/worker/package_linux64/build/src/julia_internal.h:339 [inlined]
jl_gc_alloc at /buildworker/worker/package_linux64/build/src/gc.c:3290
_new_array_ at /buildworker/worker/package_linux64/build/src/array.c:127 [inlined]
_new_array at /buildworker/worker/package_linux64/build/src/array.c:193 [inlined]
jl_alloc_array_1d at /buildworker/worker/package_linux64/build/src/array.c:464
jl_exprn at /buildworker/worker/package_linux64/build/src/builtins.c:1279
jl_copy_ast at /buildworker/worker/package_linux64/build/src/ast.c:966
jl_copy_ast at /buildworker/worker/package_linux64/build/src/ast.c:970
jl_expand_with_loc_warn at /buildworker/worker/package_linux64/build/src/ast.c:1248
jl_toplevel_eval_flex at /buildworker/worker/package_linux64/build/src/toplevel.c:667
jl_toplevel_eval_flex at /buildworker/worker/package_linux64/build/src/toplevel.c:830
jl_toplevel_eval_in at /buildworker/worker/package_linux64/build/src/toplevel.c:944
[...]
```

**Comment by odow** [comment]
I get a `Page not found` error: https://github.com/jump-dev/KNITRO.jl/blob/master/test/knitroapi.jl#L450-L541.

What's the test set? Hard to debug if it is non-deterministic.

**Comment by odow** [comment]
Closed by #238
